algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i

```diff
@@ -0,0 +1,286 @@
+services::Status TrainBatchKernel<algorithmFPType, method, cpu>::compute(
+    const HostAppIfacePtr & pHost, const NumericTablePtr & x, const NumericTablePtr & y, elastic_net::Model & m, Result & res,
+    const Parameter & par, services::SharedPtr<daal::algorithms::optimization_solver::mse::Batch<algorithmFPType> > & objFunc)
+{
+    services::Status s;
+    const size_t nFeatures           = x->getNumberOfColumns();
+    const size_t nRows               = x->getNumberOfRows();
+    const size_t p                   = nFeatures + 1;
+    const size_t nDependentVariables = m.getBeta()->getNumberOfRows();
+    DAAL_ASSERT(p == m.getNumberOfBetas());
+
+    algorithmFPType * xMeansPtr;
+    daal::internal::TArray<algorithmFPType, cpu> xMeans;
+
+    algorithmFPType * yMeansPtr;
+    daal::internal::TArray<algorithmFPType, cpu> yMeans;
+
+    NumericTablePtr xTrain = x;
+    NumericTablePtr yTrain = y;
+    if (par.interceptFlag == true)
+    {
+        if (par.dataUseInComputation == doNotUse)
+        {
+            int result = 0;
+            xTrain     = daal::internal::HomogenNumericTableCPU<algorithmFPType, cpu>::create(nFeatures, nRows, &s);
+            DAAL_CHECK_STATUS_VAR(s);
+            yTrain = daal::internal::HomogenNumericTableCPU<algorithmFPType, cpu>::create(nDependentVariables, nRows, &s);
+            DAAL_CHECK_STATUS_VAR(s);
+            daal::internal::WriteRows<algorithmFPType, cpu> xTrainBD(xTrain.get(), 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(xTrainBD);
+            daal::internal::WriteRows<algorithmFPType, cpu> yTrainBD(yTrain.get(), 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(yTrainBD);
+            algorithmFPType * xTrainPtr = xTrainBD.get();
+            algorithmFPType * yTrainPtr = yTrainBD.get();
+
+            daal::internal::WriteRows<algorithmFPType, cpu> xBD(x.get(), 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(xBD);
+            daal::internal::WriteRows<algorithmFPType, cpu> yBD(y.get(), 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS(yBD);
+            algorithmFPType * xPtr = xBD.get();
+            algorithmFPType * yPtr = yBD.get();
+            result |= daal::services::internal::daal_memcpy_s(xTrainPtr, nFeatures * nRows * sizeof(algorithmFPType), xPtr,
+                                                              nFeatures * nRows * sizeof(algorithmFPType));
+            result |= daal::services::internal::daal_memcpy_s(yTrainPtr, nDependentVariables * nRows * sizeof(algorithmFPType), yPtr,
+                                                              nDependentVariables * nRows * sizeof(algorithmFPType));
+            DAAL_CHECK(!result, services::ErrorMemoryCopyFailedInternal);
+        }
+
+        daal::internal::WriteRows<algorithmFPType, cpu> xBD(xTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(xBD);
+        daal::internal::WriteRows<algorithmFPType, cpu> yBD(yTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(yBD);
+        algorithmFPType * xPtr = xBD.get();
```

Comment:
algorithmFPType * const

and similar bellow

Reply:
it was fixed, yes it's like a cherry-pick of commit: 5c4178b